src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -687,6 +687,22 @@ public async Task DismissCompletionSessionsAsync(CancellationToken cancellationT
         broker.DismissAllSessions(view);
     }
 
+    public async Task WaitForCompletionSessionsAsync(CancellationToken cancellationToken)
+    {
+        await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+        await WaitForCompletionSetAsync(cancellationToken);
+
+        var view = await GetActiveTextViewAsync(cancellationToken);
+        var broker = await GetComponentModelServiceAsync<ICompletionBroker>(cancellationToken);
+        var sessions = broker.GetSessions(view);
+        while (sessions.Count == 0)
```

Comment:
📝 Note that it is generally not acceptable to wait in this fashion during integration tests. It is both fundamentally flaky since it doesn't account for the possibility that more than one operation complete during the wait, and slow since it fails to immediately proceed with the test code when the view rendering is complete.